src/utils.cpp

```diff
@@ -109,3 +130,163 @@ KeyRingReceive ZypperUtils::keyReport()
     return s_keyReceiveReport;
 }
 
+/************************************* ZYPPER INSTALL ************************************/
+void ZypperUtils::markPackagesForInstallation( const string& repoUrl, const string& packageName )
+{
+    Pathname sysRoot( "/" );
+    // acquire initial zypp lock
+    s_zypp = getZYpp(); 
+    
+    makeNecessaryChangesToRunAsRoot( sysRoot );
+    initTarget( sysRoot );
+    initRepoManager( repoUrl, packageName );
+    
+    //TRUST_AND_IMPORT_KEY - we are adding this to the system repos
+    KeyRing::setDefaultAccept( KeyRing::TRUST_AND_IMPORT_KEY );
+    cout << "TRUST_AND_IMPORT_KEY accepted" << endl;
+    refreshRepoManager();
```

Comment:
`markPackagesForInstallation( const string&, const string& )` is invoked only after the user feeds in the **su** password (`xdg-su -u root -c \"/usr/sbin/oneclickhelper`)

Point taken about `KeyRing::setDefaultAccept`. Thank you for your time.